torch/csrc/jit/codegen/cuda/kernel_ir_dispatch.cpp

```diff
@@ -17,6 +16,48 @@ T* ptr(T* obj) {
   return obj;
 }
 
+template <typename T>
+void Statement::kirDispatch(T handler, Statement* stmt) {
+  if (stmt->isVal()) {
+    ptr(handler)->handle(stmt->as<kir::Val>());
+  } else if (stmt->isExpr()) {
+    ptr(handler)->handle(stmt->as<kir::Expr>());
+  } else
+    TORCH_INTERNAL_ASSERT(false, "Unknown stmttype in dispatch!");
+}
+
+template <typename T>
+void Statement::kirConstDispatch(T handler, const Statement* stmt) {
+  if (stmt->isVal()) {
+    ptr(handler)->handle(stmt->as<kir::Val>());
+  } else if (stmt->isExpr()) {
+    ptr(handler)->handle(stmt->as<kir::Expr>());
+  } else
+    TORCH_INTERNAL_ASSERT(false, "Unknown stmttype in dispatch!");
+}
+
+template void Statement::kirDispatch(kir::OptOutDispatch, Statement*);
```

Comment:
nit: should this be `kir::OptOutDispatch&`?

Reply:
Heh, that seems to be in dispatch.cpp as well. I think you're right, surprised it never caused issues.

Reply:
I'm going to fix this in https://github.com/csarofeen/pytorch/pull/1353